File: athrill-account-client/projects/at-models/src/lib/at-datas/at-data-type.model.ts

import { BaseModel } from '../at-base.model';
import { ATDataValueModel } from './at-data-value.model';

export class ATDataTypeModel extends BaseModel
{
    public ATDataTypeId : number;
    public NameKey : string;
    public DisplayName : string;
    public Description : string;
    public IsActive : boolean;

    public ATDataValues :  ATDataValueModel[];

    public Id : number;
}

File: athrill-account-server/Libraries/Model/ATDatas/ATDataTypeModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AT.Model.ATDatas
{
    public class ATDataTypeModel : BaseModel
    {
        public int ATDataTypeId { get; set; }
        public string NameKey { get; set; }
        public string DisplayName { get; set; }
        public string Description { get; set; }
        public bool IsActive { get; set; }


        public  ICollection<ATDataValueModel> ATDataValues { get; set; }

        public override int Id => ATDataTypeId;
        public override string ToString()
        {
            return DisplayName;
        }
    }
}
